language/move-vm/runtime/src/runtime.rs

```diff
@@ -106,11 +106,21 @@ impl VMRuntime {
         // load the script, perform verification
         let (main, type_params) = self.loader.load_script(&script, &ty_args, data_store)?;
 
-        // build the arguments list for the main and check the arguments are of restricted types
-        let first_param_opt = main.parameters().0.get(0);
-        if first_param_opt.map_or(false, |sig| is_signer_reference(sig)) {
-            args.insert(0, Value::transaction_argument_signer_reference(sender))
+        // Build the arguments list for the main and check the arguments are of restricted types.
+        // Signers are built up from left-to-right. Not all signer arguments passed in may be used (e.g., if
+        // the script doesn't take a &signer, or if it isn't an admin script with one &signer).
+        let parameters = &main.parameters().0;
```

Comment:
This is... a bit peculiar. It feels brittle in how it would work out. 

Would it be possible to take all or nothing here? Either no signers, or you need a signer for every sender

Reply:
We could take all or nothing. One reason I preferred this is because this allows reusing "normal" scripts in an admin script context. Happy to go either way, but I do have a _slight_ preference for this "prefix" semantics.

Reply:
Lets see how other people feel
@dariorussi @sblackshear any thoughts?

Reply:
All-or-nothing seems like a good discipline to me (but we can always fix it once we allow "true" multisigner txes without forging).